File: fixture-app-rust/tests/divide_test.rs
use revi_fixture_app_rust::divide::divide_share;

#[test]
fn divide_share_returns_zero_when_parts_is_zero() {
    assert_eq!(divide_share(100, 0), 0);
}

#[test]
fn divide_share_splits_evenly() {
    assert_eq!(divide_share(100, 4), 25);
}

File: fixture-app-node/divide.test.js
const { divideShare } = require("./divide");

describe("divideShare", () => {
  test("returns null instead of throwing when parts is zero", () => {
    expect(() => divideShare(100, 0)).not.toThrow();
    expect(divideShare(100, 0)).toBeNull();
  });

  test("returns null when parts is negative", () => {
    expect(divideShare(100, -5)).toBeNull();
  });

  test("divides evenly when parts is positive", () => {
    expect(divideShare(100, 4)).toBe(25);
  });
});
